src/vivarium/framework/population/manager.py

```diff
@@ -145,12 +145,30 @@ def private_columns(self) -> pd.DataFrame:
             raise PopulationError("Population has not been initialized.")
         return self._private_columns
 
+    @overload
     def get_private_columns(
         self,
         component: Component,
         index: pd.Index[int] | None = None,
-        columns: str | Sequence[str] | None = None,
```

Comment:
mypy was throwing overload collisions since evidently a str is a Sequence (and a typing.Collection), so let's just explicitly say it's a tuple or list.